control-plane/connect-inject/annotations.go

```diff
@@ -86,6 +86,10 @@ const (
 	annotationConsulSidecarMemoryLimit   = "consul.hashicorp.com/consul-sidecar-memory-limit"
 	annotationConsulSidecarMemoryRequest = "consul.hashicorp.com/consul-sidecar-memory-request"
 
+	// annotations for sidecar volumes.
+	annotationConsulSidecarUserVolume      = "consul.hashicorp.com/consul-sidecar-user-volume"
```

Comment:
clarifying q for my understanding: these will reference a volume that exists on the pod? and then mount that to the sidecar container as well?

Reply:
Correct!
The user will need to either:
* specify both annotations, which would add a volume to the pod + mount to the sidecar.
* specify only the volumeMount annotation and already have the volume defined in the pod's spec.